bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/handler/ZoneHandler.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.alarmdecoder.internal.handler;
+
+import static org.openhab.binding.alarmdecoder.internal.AlarmDecoderBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.alarmdecoder.internal.config.ZoneConfig;
+import org.openhab.binding.alarmdecoder.internal.protocol.ADMessage;
+import org.openhab.binding.alarmdecoder.internal.protocol.EXPMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ZoneHandler} is responsible for handling wired zones (i.e. REL & EXP messages).
+ *
+ * @author Bob Adair - Initial contribution
+ * @author Bill Forsyth - Initial contribution
+ */
+@NonNullByDefault
+public class ZoneHandler extends ADThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ZoneHandler.class);
+
+    private @NonNullByDefault({}) ZoneConfig config;
```

Comment:
Ok. Easy enough to fix. It's a bit silly, though, isn't it? Now in every handler we're just creating a dummy config object in the constructor that is then immediately thrown away and replaced with the real one in the initialize() method. The other alternatives seem even more annoying, though.

Reply:
A little silly yes, but it is preferable to NPEs. The null checker may have it's annoyances but I've generally grown to like it since helps address possible concurrency issues as well as preventing NPEs. I just really wish it handled maps properly.